raquet/geotiff2raquet.py

```diff
@@ -210,6 +218,34 @@ def read_statistics(
     )
 
 
+def read_statistics_numpy(values: "numpy.array", nodata: int | float | None):
+    """Calculate statistics for array of numeric values and optional nodata value"""
+    if nodata is not None:
+        masked_values = numpy.ma.masked_array(values, values == nodata)
+        value_count = int(masked_values.count())
+    else:
+        masked_values = values
+        value_count = values.size
+
+    if value_count == 0:
+        return NoDataStats()
+
+    if masked_values.dtype in (numpy.float16, numpy.float32, numpy.float64):
+        ptype = float
+    else:
+        ptype = int
+
+    return RasterStats(
+        count=value_count,
+        min=ptype(masked_values.min()),
+        max=ptype(masked_values.max()),
+        mean=float(masked_values.mean()),
+        stddev=float(masked_values.std()),
+        sum=ptype(masked_values.sum()),
+        sum_squares=float((masked_values.astype(ptype) ** 2).sum()),
```

Comment:
Why are you casting twice in `sum_squares`? I think it's better to do it in the same way as with `sum`:
`sum_squares=ptype((masked_values ** 2).sum()),`

Reply:
Casting via `astype()` is needed to allow `uint8` Numpy arrays to work, without this the result values are all modulo 256.

The outer cast ensures that the final value can be used in JSON without throwing a TypeError.